toolchest_client/tools/tool.py

```diff
@@ -322,25 +339,49 @@ def _check_thread_health(self):
                 self._kill_query_threads()
                 raise ToolchestException("A job irrecoverably failed. See logs above for details.")
 
-    def _wait_for_threads_to_finish(self, check_health=True, wait_for_upload=True):
-        """Waits for all jobs and their corresponding threads to finish while printing their statuses."""
-        elapsed_seconds = 0
+    def _start_or_check_stream(self):
+        # Start streaming (in place of regular status updates).
+        if not self.streaming_asyncio_task:
+            # Clear the last status line before streaming begins
+            print("Printed lines:".ljust(120))
```

Comment:
Is all output streaming prefaced with `Printed lines:`?

If so, I don't think we should. It's the equivalent of using tmpenv with a banner at the top saying "remote environment".